### DISCUSSION TITLE
How to add metrics to scraper agent dynamically?

### DISCUSSION THREAD

**Discussion by SurveySky** [tomkerkhove/promitor#2037] [Q&A] [answered]
Is there a way to generate all metrics for resources dynamically? I want to have all metrics for defined resource discovery groups and would love to implement dynamic metric scraping configuration. Azure has metrics definition list [here ](https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-supported) but there is no way to get these definitions without providing a specific resourceUri. Maybe I'm missing something. If that is not possible maybe there is some kind of default metrics I could scrape without having to specify them by hand?

**Comment by tomkerkhove** [comment, marked as answer]
No you cannot. You have to define the metrics that you want to get for a given resource or discovery group.

This is by design given ARM has low throttling limitations and we don't want end-users to be too greedy.

The idea is indeed to use https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-supported to determine what metrics you really need, add it to the metric declaration and scrape them.

**Reply by tomkerkhove** [reply]
Otherwise you'll very quickly hit the throttling limitation.

**Reply by tomkerkhove** [reply]
Opened https://github.com/tomkerkhove/promitor/issues/2038 but not planning atm.

Feel free to give a 👍
